features/step_definitions/kata.rb

```diff
@@ -1,229 +1,103 @@
-      project(kata_ns)
-    end
+Then /^Kata runtime installed on selected worker nodes for #{QUOTED} kataconfig$/ do |kataconfig_name|
+  step %Q|I wait until number of completed kata runtime nodes for "#{kataconfig_name}" matches|
+end
 
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-    step %Q|I obtain test data file "kata/release-#{cb.master_version}/kataconfiguration_v1_kataconfig.yaml"|
-    @result = user.cli_exec(:apply, f: 'kataconfiguration_v1_kataconfig.yaml')
-    raise "Failed to apply kataconfig" unless @result[:success]
-    step %Q/I wait for "#{kata_config_name}" install to start/
-  else
-    logger.info("There's already an existing 'kataconfig' resuing it...")
-    project(kata_ns)
-    step %Q/I switch to cluster admin pseudo user/
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-  end
-  logger.info("Using kata image: #{pod.container_specs.first.image}")
-  step %Q/I wait until number of completed kata runtime nodes match for "#{kata_config_name}"/
+When /^I apply #{QUOTED} pod in #{QUOTED} namespace$/ do |pod_name, kata_ns|
+  step %Q|I obtain test data file "kata/#{pod_name}.yaml"|
+  @result = user.cli_exec(:apply, f: "#{pod_name}.yaml", n:kata_ns)
+  raise "Failed to deploy a pod" unless @result[:success]
 end
 
-Given /^I extract the channel information from subscription and save it to the#{OPT_SYM} clipboard$/ do |cb_name|
-  cb_name ||= :channel
-  step %Q/I store master major version in the :master_version clipboard/ if cb.master_version.nil?
-  step %Q|I obtain test data file "kata/release-#{cb.master_version}/subscription.yaml"|
-  channel = YAML.load(open('subscription.yaml')).dig('spec', 'channel')
-  logger.info("Subscription using channel: #{channel}")
-  cb[cb_name] = channel
+Then /^#{QUOTED} Pod should run using kata runtime$/ do |pod_name|
+  step %Q|I wait until "#{pod_name}" is ready|
 end
 
-Given /^I find all pods running with kata as runtime in the cluster and store them to the#{OPT_SYM} clipboard$/ do |cb_name|
-  ensure_admin_tagged
-  cb_name ||= :kata_pods
-  @result_pods = admin.cli_exec(:get, resource: "pods", all_namespaces: true, o: "jsonpath='{.items[?(@.spec.runtimeClassName==\"kata\")].metadata.name}'")
-  @result_namespaces = admin.cli_exec(:get, resource: "pods", all_namespaces: true, o: "jsonpath='{.items[?(@.spec.runtimeClassName==\"kata\")].metadata.namespace}'")
-  pods = eval(@result_pods[:response]).split(' ')
-  ns = eval(@result_namespaces[:response]).split(' ')
-  ns_pods_list = ns.zip pods
-  kata_pods = []
-  ns_pods_list.each do |ns, p|
-    project(ns)
-    pod_obj = pod(p)
-    kata_pods << pod_obj
+Given /^I wait until #{QUOTED} is ready$/ do |pod_name|
```

Comment:
@valtair1891 we already have existing steps to wait for resource to be ready.

Reply:
yes we do for kataconfig only, in addition in this step we're actually verifying the status and not just existence

Reply:
@valtair1891 if your step does more than just wait for readiness, then please change the step wording to indicate that.

Reply:
done